### ISSUE TITLE
Swagger not generated properly for @Consumes(MediaType.MULTIPART_FORM_DATA)

### ISSUE THREAD

**Issue by prateekhyland** [stoicflame/enunciate#1215]
When an endpoint is exposed as @Consumes(MediaType.MULTIPART_FORM_DATA), swagger is not generated properly. Also under Datatypes JSON, the class is not getting generated that was passed as parameter in the endpoint.

**Comment by stoicflame** [comment]
Please provide details about what you're observing and what you're expecting from the Swagger docs.

As for the JSON data type, what indicators does Enunciate have to know that the parameter to the endpoint is to be interpreted as a JSON object?

**Comment by prateekhyland** [comment]
I have an endpoint that is : 

@Path("documents")
	@POST
	@Consumes(MediaType.MULTIPART_FORM_DATA)
	DocumentStoreInformationType createDocument(@QueryParam("verify") @DefaultValue("false") boolean verify,
			CreateDocumentInputType properties)

another endpoint is : 

@Path("lookup-entry")
	@POST
	@Override
	LookupEntryReferenceType createLookupEntry(CreateLookupEntryInputType properties)

For the second one I am getting docs generated as given in the image attached.. But not getting similar one for CreateDocumentInputType
![ss](https://github.com/user-attachments/assets/9afa54a3-3ea8-4a76-976a-92e0af63fffe)

**Comment by stoicflame** [comment]
> For the second one I am getting docs generated as given in the image attached.. But not getting similar one for CreateDocumentInputType

Yes the `createLookupEntry` method "consumes" `application/json`, so Enunciate knows that `CreateLookupEntryInputType` is supposed to be serialized as JSON. The `createDocument` method consumes `multipart/form-data`, so Enunciate only knows that `CreateDocumentInputType` is supposed to be serialized as a string value and there's no way of knowing whether that string value should be a json string, yaml string, xml string, or something else entirely.

Am I missing something?

Also, I still need details about the reported problem of "swagger is not generated properly."

**Comment by stoicflame** [comment]
Still waiting on some clarification, but I'm going to assume this has been addressed by the fix for #1228.

You're welcome to re-open as needed.

**Comment by stoicflame** [comment]
Enunciate `2.19.0` has been released.
